stglib/core/utils.py

```diff
@@ -1004,40 +1004,84 @@ def create_z(ds):
     ds["z"].attrs["units"] = "m"
     ds["z"].attrs["standard_name"] = "height"
 
-    if "P_1ac" in ds:
-        if "bindist" in ds:
-            if ds.attrs["orientation"].upper() == "DOWN":
-                presvar = np.nanmean(ds["P_1ac"]) + ds["bindist"].values
-            elif ds.attrs["orientation"].upper() == "UP":
-                presvar = np.nanmean(ds["P_1ac"]) - ds["bindist"].values
-        else:
-            presvar = np.nanmean(ds["P_1ac"])
-    elif "P_1" in ds:
-        if "bindist" in ds:
-            if ds.attrs["orientation"].upper() == "DOWN":
-                presvar = np.nanmean(ds["P_1"]) + ds["bindist"].values
-            elif ds.attrs["orientation"].upper() == "UP":
-                presvar = np.nanmean(ds["P_1"]) - ds["bindist"].values
-        else:
-            presvar = np.nanmean(ds["P_1"])
-    else:
-        if "bindist" in ds:
-            if ds.attrs["orientation"].upper() == "DOWN":
-                presvar = ds.attrs["WATER_DEPTH"] + ds["bindist"].values
-            elif ds.attrs["orientation"].upper() == "UP":
-                presvar = ds.attrs["WATER_DEPTH"] - ds["bindist"].values
+    # find depth dimension values
+    # check for sea floor depth standard names
+    sfds = ["geoid", "geopotential_datum", "mean_sea_level", "reference_ellipsoid"]
+    for k in sfds:
+        name = "sea_floor_depth_below_" + k
+        longname = "depth below " + k.replace("_", " ")
+        if name in ds.attrs:
+            if "bindist" in ds:
+                if ds.attrs["orientation"].upper() == "DOWN":
+                    depvar = (
+                        ds.attrs[name]
+                        - ds.attrs["initial_instrument_height"]
+                        + ds["bindist"].values
+                    )
+                elif ds.attrs["orientation"].upper() == "UP":
+                    depvar = (
+                        ds.attrs[name]
+                        - ds.attrs["initial_instrument_height"]
+                        - ds["bindist"].values
+                    )
+            else:
+                depvar = ds.attrs[name] - ds.attrs["initial_instrument_height"]
+            break
+
+    if "depvar" not in locals():
```

Comment:
Then replace this line with `if not depvar:`

Reply:
instead used -->  if depvar is None: